extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java

```diff
@@ -423,73 +422,85 @@ private static void applyOpenshiftResources(OpenShiftClient client, List<HasMeta
         }
     }
 
-    private static void openshiftBuild(OpenShiftClient client, List<HasMetadata> buildResources, File binaryFile,
-            OpenshiftConfig openshiftConfig) {
+    private static void openshiftBuild(List<HasMetadata> buildResources, File binaryFile,
+            OpenshiftConfig openshiftConfig, KubernetesClientBuilder kubernetesClientBuilder) {
         distinct(buildResources).stream().filter(i -> i instanceof BuildConfig).map(i -> (BuildConfig) i)
-                .forEach(bc -> openshiftBuild(client, bc, binaryFile, openshiftConfig));
+                .forEach(bc -> {
+                    Build build = startOpenshiftBuild(bc, binaryFile, openshiftConfig, kubernetesClientBuilder);
+                    waitForOpenshiftBuild(build, openshiftConfig, kubernetesClientBuilder);
+                });
     }
 
     /**
      * Performs the binary build of the specified {@link BuildConfig} with the given
      * binary input.
      *
-     * @param client The openshift client instance
      * @param buildConfig The build config
      * @param binaryFile The binary file
      * @param openshiftConfig The openshift configuration
+     * @param kubernetesClientBuilder The kubernetes client builder
      */
-    private static void openshiftBuild(OpenShiftClient client, BuildConfig buildConfig, File binaryFile,
-            OpenshiftConfig openshiftConfig) {
-        Build build;
-        try {
-            build = client.buildConfigs().withName(buildConfig.getMetadata().getName())
-                    .instantiateBinary()
-                    .withTimeoutInMillis(openshiftConfig.buildTimeout.toMillis())
-                    .fromFile(binaryFile);
-        } catch (Exception e) {
-            Optional<Build> running = runningBuildsOf(client, buildConfig).findFirst();
-            if (running.isPresent()) {
-                LOG.warn("An exception: '" + e.getMessage()
-                        + " ' occurred while instantiating the build, however the build has been started.");
-                build = running.get();
-            } else {
-                throw openshiftException(e);
+    private static Build startOpenshiftBuild(BuildConfig buildConfig, File binaryFile,
+            OpenshiftConfig openshiftConfig, KubernetesClientBuilder kubernetesClientBuilder) {
+        try (KubernetesClient kubernetesClient = kubernetesClientBuilder.build()) {
+            OpenShiftClient client = toOpenshiftClient(kubernetesClient);
+            try {
+                return client.buildConfigs().withName(buildConfig.getMetadata().getName())
+                        .instantiateBinary()
+                        .withTimeoutInMillis(openshiftConfig.buildTimeout.toMillis())
+                        .fromFile(binaryFile);
+            } catch (Exception e) {
+                Optional<Build> running = buildsOf(client, buildConfig).stream().findFirst();
```

Comment:
I also changed `runningBuildsOf` by `buildsOf` because there is a chance that the build is completed or failed when calling this method. The state is properly read in the next method.
The method `runningBuildsOf` was removed because there were no more usages.